### PR TITLE
switched all scripts/shebangs using python2.7 to python3.x

### LINKED ISSUE #19795: Codebase contains lingering legacy python code
#### Problem

We still have some python2.7 in the following scripts.  Python2 is EOL
```
ci/dependabot-pr.sh
system-test/automation_utils.sh
```

#### Proposed Solution

Port them to Python3

### PR DESCRIPTION
**Pull request by drewtaylors**
#### Problem
Python2.7 is EoL

#### Summary of Changes
Modified all scripts to use `python3` instead of `python` (2.7)
Verified all shebangs in python files use `python3` instead of `python` (2.7)

Fixes #19795

### CODE REVIEW

**Comment by t-nelson** [comment]
Thanks for taking this on!

Were the scripts all verified to be syntactically correct under python3?

**Comment by drew-fisker** [comment]
@t-nelson everything should be good with the scripts now!

* Verified and validated all python files that are being called by shell scripts are being called with `python3` or using a `python3` shebang
* Ensured all files missing the `python3` shebang were switched over and validated successful compile
* Fixed my overlook of in-line python in the automation shell script

**Comment by drewtaylors** [comment]
@t-nelson everything should be good with the scripts now!

* Verified and validated all python files that are being called by shell scripts are being called with python3 or using a python3 shebang
* Ensured all files missing the python3 shebang were switched over and validated successful compile
* Fixed my overlook of in-line python in the dependabot shell script

File: ci/dependabot-pr.sh [hunk 1]
```diff
@@ -23,7 +23,7 @@ echo --- "(FAILING) Backpropagating dependabot-triggered Cargo.lock updates"
 name="dependabot-buildkite"
 api_base="https://api.github.com/repos/solana-labs/solana/pulls"
 pr_num=$(echo "$BUILDKITE_BRANCH" | grep -Eo '[0-9]+')
-branch=$(curl -s "$api_base/$pr_num" | python -c 'import json,sys;print json.load(sys.stdin)["head"]["ref"]')
+branch=$(curl -s "$api_base/$pr_num" | python3 -c 'import json,sys;print(json.load(sys.stdin)["head"]["ref"])')
```
Comment (ryoqun) [inline ci/dependabot-pr.sh:26]: :eyes:

**Comment by t-nelson** [comment]
lgtm.  Thanks for contributing!

**Pull request merged**